### DISCUSSION TITLE
Suggested improvements

### DISCUSSION THREAD

**Discussion by sam9090** [OOlashyn/PCF-AttachmentGalleryControl#16] [Ideas]
Hello Oleksandr.
AttachmentGallery is quite remarkable. I'm working on a project which requires image management and your control fits nicely. Thank you for making this available.

I would like to make some suggested improvements:
1. Display loading indicator when images are loading. When there are many images it can take a while to load and users might think it's not working.
2. Show the image file name (in the lightbox).
3. Provide delete feature.

Sam

**Comment by OOlashyn** [comment]
Hi Sam,
Happy that my component helped you. Those are good suggestions and I will include them in the next version. I am not sure when exactly I will release it yet, but hopefully, it will be soon.

**Comment by sam9090** [comment]
Sounds great, I look forward to it.

**Comment by Triangle333** [comment]
Hello!

Voting enthusiastically for item 2! (Show the image/pdf file name)
![image](https://user-images.githubusercontent.com/8589307/145522752-457f3b26-0f6e-43fd-be14-8f8e857bf089.png)

Plus:
4. Add sorting feature, by filenames or by dates. Fixed configuration using params will be enough.
5. Is it possible to show thumbnails of the 1st page for PDFs? Would be really cool to have this

**Reply by OOlashyn** [reply]
Hi @sam9090  and @Triangle333 ,
I had a crazy last year with a lot of things on my plate (switching jobs, relocation to a different country - to name a few), so I wasn't working on control at all. But now when things are a bit settled down I will start working on it again. Later next week first beta of 2.0 should be ready and I hope to share it with you for further testing.

**Comment by ahgobjovic** [comment]
Hi @OOlashyn  
Thank you for this component I think it is very useful. 
I have just imported it on my CRM, but I can't set it up. I've just imported it into my CRM, but I can't set it up. 
can you give me the steps please ?

Thanks

**Reply by OOlashyn** [reply]
Hi @ahgobjovic ,
Check the Wiki page for the installation instructions.
